### ISSUE TITLE
Crashes on startup - scanning ~/.ladspa dir

### ISSUE THREAD

**Issue by Lucretia** [hydrogen-music/hydrogen#1591]
Hydrogen version *          : HEAD @ 2109 hrs BST 23/02/2022
Operating system + version  : Gentoo (testing)
Audio driver + version      : Pipewire-0.3.51-r1 (Jack)

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Starting up from console:

```
QT_AUTO_SCREEN_SCALE_FACTOR=1.5 hydrogen -VDebug
...
(I) ::int main(int, char**) Using QT version 5.15.4
(I) ::int main(int, char**) Using data path: /usr/share/hydrogen/data/
(I) ::int main(int, char**) Using locale: /usr/share/hydrogen/data/i18n/
(I) Hydrogen::Hydrogen [Hydrogen]
(I) Effects::getPluginList *** [getPluginList] reading directory: /home/laguest/.ladspa
[main] Unknown exception X-(
```

Changing the files inside that dir to another extension, gives the same error, moving that dir allows the program to run as does moving the two plugin's out of that directory, they are amp.so and delay.so.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

*  H2 version : 1.1.1

**Comment by theGreatWhiteShark** [comment]
Hey @Lucretia,

Seems like your to plugins `amp.so` and `delay.so` are bricked and causing the crash. Do they work in other applications?

**Comment by Lucretia** [comment]
Seems they break guitarix as well. Both applications should catch any error, not crash without giving any indication of what happened and that's what both do.

**Comment by theGreatWhiteShark** [comment]
> Both applications should catch any error, not crash without giving any indication of what happened and that's what both do.

I'm not an expert on these system-level kind of thing but it [looks like](https://github.com/hydrogen-music/hydrogen/blob/master/src/gui/src/main.cpp#L74) we are already doing this.

**Comment by Lucretia** [comment]
Well, I've installed the latest HEAD version, so that's not catching this error.

**Comment by Lucretia** [comment]
Here are the plugins which kills it (place inside ~/.ladspa):

[dot-ladspa.tar.gz](https://github.com/hydrogen-music/hydrogen/files/8782440/dot-ladspa.tar.gz)

**Comment by cme** [comment]
Well.... There's not really a lot we can or indeed *should* do to recover from an error like this -- as soon as we've executed some potentially bad code from a plugin, there's no way to tell what damage that might have done to the process, so stopping more or less immediately is the right thing to do.

(We don't isolate ladspa plugins in a separate host process, they're executed as part of the hydrogen process. Changing this would be... a pain)

We certainly could try and emit some more helpful messages in the case of such a crash, though.

**Comment by elpescado** [comment]
> Seems they break [guitarix](https://github.com/brummer10/guitarix/issues/112) as well. Both applications should catch any error, not crash without giving any indication of what happened and that's what both do.

I think that's because of how LADSPA architecture - host application and plugins are so intertwined that when plugin crashes there's nothing really that could be done to recover host.

**Comment by Lucretia** [comment]
You can check if the load failed, and log it if it did. Then check if on initialisation of the plugin failed, and then log that too. Having *some* indication of what happened is better than nothing.

**Comment by cme** [comment]
> You can check if the load failed, and log it if it did. Then check if on initialisation of the plugin failed, and then log that too. Having *some* indication of what happened is better than nothing.

Yep, agreed

**Comment by Lucretia** [comment]
Although, if the app is going to kill itself, I'd prefer to be given a message box saying why if possible.

**Comment by cme** [comment]
> Although, if the app is going to kill itself, I'd prefer to be given a message box saying why if possible.

Yeah, this is also an issue -- errors go to the logs, not so much to anything the user will see unless they're looking at the console.

**Comment by Lucretia** [comment]
> > Although, if the app is going to kill itself, I'd prefer to be given a message box saying why if possible.
> 
> Yeah, this is also an issue -- errors go to the logs, not so much to anything the user will see unless they're looking at the console.

But a user who is faced with a silent crash, is going to want to know WTF just happened, message box. This is usability basics.

**Comment by cme** [comment]
Indeed. Captured in #1592.

**Comment by theGreatWhiteShark** [comment]
> Although, if the app is going to kill itself, I'd prefer to be given a message box saying why if possible.

Hey @Lucretia ,

could you check whether the newest version of Hydrogen on `master` serves your needs?
